src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -128,6 +109,106 @@ class Target
             await VerifyAnalyzerCSAsync(source, tfmAndOption);
         }
 
+        [Fact]
+        public async Task GeneratedFile_RazorWarns()
+        {
+            var csSource = @"
+#pragma checksum ""D:\samples\blazor\Pages\Counter.razor"" ""{8829d00f-11b8-4213-878b-770e8597ac16}"" ""1abc50bc4b6da164a335c3b53c2b8f2accc8a7af01eb264797d0852f16989494""
+// <auto-generated type=""Razor""/>
+#pragma warning disable 1591
+namespace blazor.Pages
+{
+    #line hidden
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+#line 1 ""D:\samples\blazor\_Imports.razor""
+using System.Net.Http;
+
+#line default
+#line hidden
+#line 2 ""D:\samples\blazor\_Imports.razor""
+using System.Net.Http.Json;
+
+#line default
+#line hidden
+    public partial class Counter  // 23
+    {
+
+#line 11 ""D:\samples\blazor\Pages\Counter.razor""
+       
+    private int currentCount = 0;
+
+    private void IncrementCount()
+    {
+        Thread thread = new Thread(T1);
+        [|thread.Start()|];
+        currentCount++;
+    }
+
+    void T1 () {
+        Console.WriteLine(""Thread started"");
+    }
+
+#line default
+#line hidden
+    }
+}
+#pragma warning restore 1591";
+            await VerifyAnalyzerCSAsync(csSource, s_msBuildPlatforms);
+        }
+
+        [Fact]
+        public async Task GeneratedFile_NotRazorNotWarn()
```

Comment:
Can we also added VB tests for the same?

Reply:
I have tried to get generated file in VB, but its not supported in blazor or ASP NetCore, though I could just create a fake one and use it